src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1278,6 +1278,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="CrossPlatformExecutable" />
   </ItemGroup>
 
+  <!-- Native AOT compilation -->
+  <!-- Native AOT properties should be shown by default for projects targeting .NET 8 or higher, except for WPF and WinForms projects -->
+  <PropertyGroup>
+    <ShowNativeAOTProperties Condition="'$(ShowNativeAOTProperties)' == ''
+                             and '$(UseWPF)' != 'true'
+                             and '$(UseWindowsForms)' != 'true'
+                             and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                             and $([MSBuild]::VersionGreaterThanOrEquals('$(_TargetFrameworkVersionWithoutV)', '8.0'))">true</ShowNativeAOTProperties>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(ShowNativeAOTProperties)' == 'true'">
+    <ProjectCapability Include="NativeAOTProperties"/>
```

Comment:
The name `NativeAOTProperties` implies that the capability is only about showing certain properties in the UI. Would it instead make sense to have a more generic name such as "SupportsNativeAOT" and show the properties based on that? Or do we need to make a distinction between projects that support Native AOT and those that should show the properties?

Reply:
IMO, "SupportsNativeAOT" could be interpreted so many different ways. I'm not sure it would be a good name for the ProjectCapability. For example, it could mean that the project successfully publishes to Native AOT and works correctly, which is not true for many (most?) apps.

"Available" or "Capable" might be better qualifiers.

Note if we rename the ProjectCapability, we should also rename the property `ShowNativeAOTProperties` to match.

Reply:
I think we can make the distinction here between showing the properties and actually supporting AOT compilation (which I think it's covered by other properties like `PublishAot`, `IsAotCompatible`, `EnableAotAnalyzer`, etc.).

Reply:
I think the "Capable" part is implied by the `ProjectCapability` item type. 😀 My preference would actually be to just call the capability `NativeAOT`, but I'd be fine with `NativeAOTAvailable` or `NativeAOTCapable`.

Reply:
> Note if we rename the ProjectCapability, we should also rename the property ShowNativeAOTProperties to match.

Or maybe just get rid of the property?

Reply:
> Or maybe just get rid of the property?

See the top comment:

> Other project types, like MAUI and Web projects, can disable this capability by setting the ShowNativeAOTProperties property to false in their props file. 

Would you rather them `<ProjectCapability Remove="NativeAOTProperties"/>`?

Reply:
That makes sense, let me get rid of the property and rename the `ProjectCapability`.